Lab_1/IK-01_Smolij_Natan/my_lib.c

```diff
@@ -0,0 +1,129 @@
+#include "my_lib.h"
+
+char update_data(char mode,sub_chunk* wav_file)
+{
+	uint16_t bps;
+	uint32_t freq;
+	memcpy(&bps,wav_file[0].data+14,2);
+	memcpy(&freq,wav_file[0].data+4,4);
+	bps/=8;
+	float* par;
+	switch(mode)
+	{
+		//multiplies
+		case 0:
+			par = malloc(sizeof(float));
+			printf("Enter multiplier[0.1:10]:");
+			scanf("%f",par);
+			if(!(par[0]>=0.1 && par[0]<=10))
+				return 1;
+		break;
+
+		//adds absolute value
+		case 1:
+			par = malloc(sizeof(float));
+			printf("Enter add value[0:65535]:");
+			scanf("%f",par);
+			if(!(par[0]>=0 || par[0]<=65535))
+				return 1;
+		break;
+
+		//trembling effect
+		case 2:
+			par = calloc(2,sizeof(float));
+			printf("Enter amplitude[0:65535]:");
+			scanf("%f",par);
+			printf("Enter frequency(>0):");
+			scanf("%f",par+1);
+			if(!(par[0]>=0 && par[0]<=65535))
+				return 1;
+			if(par[1]<0)
+				return 1;
+		break;
+
+		default:
+			printf("%d - wrong mode!\n",mode);
+			return 1;
+	}
+	int16_t* samp_data=malloc(bps);
+	//wav_file[1].chunkSize-размер сегмента чанка с данными  bps - байт на сэмпл
+	for(unsigned long int i =0 ;i<=wav_file[1].chunkSize;i+=bps)
+	{
+		memcpy(samp_data,wav_file[1].data+i,bps);
+		switch(mode)
+		{
+			case 0:
+				*samp_data=(int16_t)*samp_data*par[0];
+			break;
+
+			case 1:
+				*samp_data=(int16_t)*samp_data+par[0];
+			break;
+			case 2:
+				*samp_data=*samp_data+(int16_t)par[0]*sin(par[1]*2*3.1415926*(double)(i/bps/2 %freq)/freq);
+			break;
+		}
+		memcpy(wav_file[1].data+i,samp_data,bps);
+	}
+	free(par);
+	free(samp_data);
+}
+
+char read_wav(const char* filename)
+{
+//	printf("Attempt to open file:%s\n",filename);
+	FILE* to_rd= fopen(filename,"r");
+	if(!to_rd)
+	{
+		printf("Can`t open file:%s\n",filename);
+		return 0;
+	}
+
+	sub_chunk* wav_file= calloc(2,sizeof(sub_chunk));
```

Comment:
Please, avoid magic numbers in code. Use #define to declare all of numbers below.